### PR TITLE
Retrieve symbols on module load with nn_symbol

### PR DESCRIPTION
**Pull request by comick**
See http://nanomsg.org/development-bindings.html, this is the reccomanded way to retrieve symbols.

Not sure about:
EXPORT_CONSTANT(exports, EISCONN); // not in symbol.c, but is in nn.h
// Polling - these aren't in symbol.c but they are in nn.h
EXPORT_CONSTANT(exports, NN_POLLIN);
EXPORT_CONSTANT(exports, NN_POLLOUT);

why they were defined as symbol in node module?
They are not returned by nn_symbol and it seems they are not needed at all.
All tests pass.

### CODE REVIEW

File: src/node_nanomsg.cc [hunk 1]
```diff
@@ -435,130 +434,17 @@ void InitAll(Handle<Object> exports) {
...
+    int index = 0;
+    int value;
+    while (true) {
+        const char *symbol_name = nn_symbol(index, &value);
+        if (symbol_name == NULL) {
+            break;
+        }
+        exports->Set(NanNew(symbol_name), NanNew<Number>(value));
+        index += 1;
+    }
```
Comment (nickdesaulniers) [inline src/node_nanomsg.cc]: Can we turn this into a for-loop, use pre-increment on the loop variable, and put `int value` in the for-loop's block?

**Comment by nickdesaulniers** [comment]
:+1: :+1: :+1: patches that remove more code than they add!  Just one fixer-upper and this is good to merge.

**Comment by nickdesaulniers** [comment]
Looks good, thanks Michele.  Make sure you comment in the issue when you fix up code review nits; otherwise the maintainer doesn't get an email that the fixes were made. (or the maintainers GH settings are messed up)

**Comment by nickdesaulniers** [comment]
published in v0.3.5

**Comment by reqshark** [comment]
wow cool :+1:

**Comment by comick** [comment]
@nickdesaulniers :+1:

**Pull request merged**